src/sector.jl

```diff
@@ -1,55 +1,64 @@
-export SUNIrrep
-
 # is this type piracy?
 const SU₃ = SU{3}
 const SU₄ = SU{4}
 const SU₅ = SU{5}
-TensorKit.type_repr(::Type{SU₃}) = "SU₃"
-TensorKit.type_repr(::Type{SU₄}) = "SU₄"
-TensorKit.type_repr(::Type{SU₅}) = "SU₅"
-Base.getindex(::TensorKit.IrrepTable, ::Type{SU{N}}) where {N} = SUNIrrep{N}
+TensorKitSectors.type_repr(::Type{SU₃}) = "SU₃"
+TensorKitSectors.type_repr(::Type{SU₄}) = "SU₄"
+TensorKitSectors.type_repr(::Type{SU₅}) = "SU₅"
+Base.getindex(::TensorKitSectors.IrrepTable, ::Type{SU{N}}) where {N} = SUNIrrep{N}
 
 Base.convert(::Type{SUNIrrep{N}}, I::NTuple{N,Int}) where {N} = SUNIrrep{N}(I)
 Base.convert(::Type{SUNIrrep{N}}, I::Vector{Int}) where {N} = SUNIrrep{N}(I)
 Base.convert(::Type{SUNIrrep{N}}, I::AbstractString) where {N} = SUNIrrep{N}(I)
-Base.IteratorSize(::Type{TensorKit.SectorValues{T}}) where {T<:SUNIrrep} = Base.IsInfinite()
+function Base.IteratorSize(::Type{TensorKitSectors.SectorValues{T}}) where {T<:SUNIrrep}
+    return Base.IsInfinite()
+end
 
-function Base.iterate(::TensorKit.SectorValues{SUNIrrep{N}}, I=ntuple(zero, N)) where {N}
-    s = SUNIrrep(I)
-    k = N - 1
-    while k > 1 && I[k] == I[k - 1]
-        k -= 1
-    end
-    I = Base.setindex(I, I[k] + 1, k)
-    for l in (k + 1):(N - 1)
-        I = Base.setindex(I, 0, l)
-    end
-    return s, I
+function Base.iterate(iter::TensorKitSectors.SectorValues{<:SUNIrrep}, i::Int=1)
+    return iter[i], i + 1
+end
+
+# linear order of sectors: use manhattan ordering of dynkin labels (0-based)
+function Base.getindex(::TensorKitSectors.SectorValues{SUNIrrep{N}}, i::Int) where {N}
+    sz = ntuple(Returns(typemax(Int)), N - 1)
+    I = TensorKitSectors.manhattan_to_multidimensional_index(i, sz)
+    dk_label = I .- 1 # dynkin labels are 0-based
+    return SUNIrrep{N}(reverse(cumsum(reverse(dk_label)))..., 0)
+end
+function TensorKitSectors.findindex(::TensorKitSectors.SectorValues{SUNIrrep{N}},
+                                    s::SUNIrrep{N}) where {N}
+    I = dynkin_label(s) .+ 1
+    sz = ntuple(Returns(typemax(Int)), N - 1)
+    return TensorKitSectors.to_manhattan_index(I, sz)
```

Comment:
If this ever turns out to take up a significant fraction of runtime, there is a faster way to generate the manhattan index when all dimensions are guaranteed to be infinite-dimensional, since you can analytically do the summations involved.